src/cpp/src/text2image/stable_diffusion_xl_pipeline.hpp

```diff
@@ -0,0 +1,345 @@
+        m_unet->reshape(num_images_per_prompt * batch_size_multiplier, height, width, m_clip_text_encoder->get_config().max_position_embeddings);
+        m_vae_decoder->reshape(num_images_per_prompt, height, width);
+    }
+
+    void compile(const std::string& device, const ov::AnyMap& properties) override {
+        m_clip_text_encoder->compile(device, properties);
+        m_clip_text_encoder_with_projection->compile(device, properties);
+        m_unet->compile(device, properties);
+        m_vae_decoder->compile(device, properties);
+    }
+
+    ov::Tensor generate(const std::string& positive_prompt,
+                        const ov::AnyMap& properties) override {
+        GenerationConfig generation_config = m_generation_config;
+        generation_config.update_generation_config(properties);
+
+        // Stable Diffusion pipeline
+        // see https://huggingface.co/docs/diffusers/using-diffusers/write_own_pipeline#deconstruct-the-stable-diffusion-pipeline
+
+        const auto& unet_config = m_unet->get_config();
+        const size_t batch_size_multiplier = do_classifier_free_guidance(generation_config.guidance_scale) ? 2 : 1;  // Unet accepts 2x batch in case of CFG
+        const size_t vae_scale_factor = m_unet->get_vae_scale_factor();
+
+        if (generation_config.height < 0)
+            generation_config.height = unet_config.sample_size * vae_scale_factor;
+        if (generation_config.width < 0)
+            generation_config.width = unet_config.sample_size * vae_scale_factor;
+        check_inputs(generation_config.height, generation_config.width);
+
+        if (generation_config.random_generator == nullptr) {
+            uint32_t seed = time(NULL);
+            generation_config.random_generator = std::make_shared<CppStdGenerator>(seed);
+        }
+
+        std::vector<float> time_ids = {static_cast<float>(generation_config.width), 
+                                       static_cast<float>(generation_config.height),
+                                       0,
+                                       0, 
+                                       static_cast<float>(generation_config.width), 
+                                       static_cast<float>(generation_config.height),
+                                       };
+        ov::Tensor add_time_ids(ov::element::f32, {batch_size_multiplier, time_ids.size()});
+        float* add_time_ids_data = add_time_ids.data<float>();
+        std::copy(time_ids.begin(), time_ids.end(), add_time_ids_data);
+
+        if (batch_size_multiplier > 1) {
+            std::copy(time_ids.begin(), time_ids.end(), add_time_ids_data + time_ids.size());
+        }
+
+        ov::Tensor add_text_embeds = m_clip_text_encoder_with_projection->infer(positive_prompt, generation_config.negative_prompt, batch_size_multiplier > 1);
+        m_clip_text_encoder->infer(positive_prompt, generation_config.negative_prompt, batch_size_multiplier > 1);
```

Comment:
users can also specify prompt2, negative_prompt2 which should be used for second text encoder.
Please, check whether second prompts are defined and use them

See https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L367-L368